zio-http/shared/src/main/scala/zio/http/Server.scala

```diff
@@ -462,15 +476,24 @@ object Server extends ServerPlatformSpecific {
 
   private final case class ServerLive(
     driver: Driver,
-    bindPort: Int,
+    // A promise used to signal the first time `install`
+    // is called on this `Server` instance.
+    private val initialInstall: Promise[Nothing, Unit],
+    // A promise that represents the port of the "started" driver
+    // or a throwable if starting the driver failed for any reason.
+    private val serverStarted: Promise[Throwable, Int],
   ) extends Server {
     override def install[R](httpApp: Routes[R, Response])(implicit
       trace: Trace,
       tag: EnvironmentTag[R],
     ): URIO[R, Unit] =
-      ZIO.environment[R].flatMap(env => driver.addApp(httpApp, env.prune[R]))
+      for {
+        _ <- initialInstall.succeed(())
+        _ <- serverStarted.await.orDie
```

Comment:
My opinion here was that is we fail to start the driver, this should be raised as a "defect" -- this seems like an unrecoverable error to me but please let me know if you disagree.